src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -530,20 +517,28 @@ void OTARequestor::OnUpdateProgressChanged(Nullable<uint8_t> percent)
     OtaRequestorServerSetUpdateStateProgress(percent);
 }
 
-void OTARequestor::UpdateDefaultProviderLocation()
+bool OTARequestor::SetDefaultProviderLocation(const ProviderLocationType & providerLocation)
 {
-    // If there is currently no cached provider location or there is no OTA update in progress, update for the next periodic query
-    if (!mProviderLocation.HasValue() || (mCurrentUpdateState == OTAUpdateStateEnum::kIdle))
+    // Provider location cannot be modified if there is an OTA update in progress
```

Comment:
I would suggest removing this whole clause. The state-related logic should be kept as compact as possible. The code that calls this method should be checking the state, not the method.

Reply:
I had thought about removing this check but given that `OTARequestor` relies on the data that sits in `mProviderLocation`, I feel it is still necessary to do some basic guards. For example, if a driver or some other entity that has access inadvertently cleared this value (without this check here), and we're mid-update, that would cause some undesirable behavior in the process.